src/metax_api/models/catalog_record.py

```diff
@@ -1480,6 +1596,14 @@ def _create_new_dataset_version(self):
                 # todo this probably does not make sense... ?
                 new_version.editor['identifier'] = old_editor['identifier']
 
+        if cumulative_state:
+            new_version.cumulative_state = cumulative_state
+            if cumulative_state == self.CUMULATIVE_STATE_YES:
```

Comment:
I think a new version can only be created as a result of changing from no/closed to yes? Changing from yes to closed does not create a version.